lib/search/pagination.rb

```diff
@@ -47,9 +56,25 @@ def pagination_object
       {
         'current_page' => [current_page, total_pages].min,
         'per_page' => ENTRIES_PER_PAGE,
-        'total_pages' => total_pages,
-        'total_entries' => total_entries
+        'total_pages' => total_viewable_pages,
+        'total_entries' => total_viewable_entries
       }
     end
+
+    def total_viewable_pages
+      [total_pages, maximum_viewable_pages].min
+    end
+
+    def maximum_viewable_pages
+      (OFFSET_LIMIT / ENTRIES_PER_PAGE.to_f).floor
+    end
+
+    def total_viewable_entries
+      [total_entries, maximum_viewable_entries].min
+    end
+
+    def maximum_viewable_entries
+      (ENTRIES_PER_PAGE * total_viewable_pages) + (ENTRIES_PER_PAGE - 1)
```

Comment:
Is calculation correct? It seems that if there’s only 1 page this would return 19

Reply:
@bshyong - Yeah, it would for sure; however, `total_entries` would return `1`, and so

```ruby
    def total_viewable_entries
      [total_entries, maximum_viewable_entries].min
    end
```

would return `1`.